frb_codegen/src/logs.rs

```diff
@@ -61,5 +60,9 @@ pub fn init_logger(path: &str, verbose: bool) -> Result<(), fern::InitError> {
         _ => panic!("only allow \"debug\" and \"info\""),
     }
 
+    std::panic::set_hook(Box::new(|m| {
+        log::error!("{}", m);
+    }));
```

Comment:
hmm I wonder why do we log error instead of letting it panic?

Reply:
With current `log::error!`, it would output error message when panic occurs, like this:
```rust
2023/03/16 10:15:15 [INFO] Phase: Validate config(s)
2023/03/16 10:15:15 [ERROR] panicked at 'assertion failed: `(left == right)`
  left: `BlockIndex(2)`,
 right: `BlockIndex(0)`: config index mismatch: order index(0) != block_index(2)', frb_codegen\src/lib.rs:83:9
```
While using ` log::error!("{}", m);` alone, there would be no log message after `2023/03/16 10:15:15 [INFO] Phase: Validate config(s)`, which is not expected.

Reply:
Well I am a bit confused. So if I understand correctly, you mean the `assert_eq!` alone does not generate good error messages, and we have to add an extra `log::error` in `set_hook`? That looks weird...

In addition, if we really want to set_hook, we should `take_hook` and call that original hook as well

Reply:
> you mean the assert_eq! alone does not generate good error messages,

All rust default `panic!` or `assert` like statements would generate an abort message, but would not log it to disk.  Using `log::error` just makes it convenient for debug logging. 

Meanwhile, customizing the panic hook(using `set_hook(...)`) does not change how the panic is propagated or handled by the runtime.  That means the panicked thread would be aborted as before.  And the panic message would still print out in the terminal for a built app using it this way--- the printed-out message is the same as that in the disk log.    (You can try it yourself.)

> if we really want to set_hook, we should take_hook and call that original hook as well, because there can be a hook there

Do you afraid that using `set_hook(...)` alone instead of `take_hook()` would lose the original information? 

Since the code is like this:
```rust
std::panic::set_hook(Box::new(|m| {
        log::error!("{}", m);
    }));
```
The original message is already in `m`, so it is not lost.  Of course, we can take it like this: 
```rust
use std::panic;

fn main() {
    // Store the current panic hook in a variable
    let old_hook = panic::take_hook();
    
    // Register a new panic hook that prints "Custom panic!" instead of the default message
    panic::set_hook(Box::new(|_| {
        println!("Custom panic!");
    }));
    
    // This will trigger the custom panic hook
    panic!("Something went wrong");
    
    // Restore the original panic hook (this line will not be executed because of the previous panic)
    panic::set_hook(old_hook);
}
```
But I don't think using `take_hook` is necessary here, since the customized log::error message contains the same content as the default one --- it just added the logging to disk feature.

Reply:
I see - you want *disk* logging, then definitely yes.

I mean,

```
    let old_hook = panic::take_hook();
    panic::set_hook(Box::new(|_| {
        println!("Custom panic!");
        old_hook(); // <--- CALL IT here
    }));
```

Reply:
> I see - you want _disk_ logging, then definitely yes.
> 
> I mean,
> 
> ```
>     let old_hook = panic::take_hook();
>     panic::set_hook(Box::new(|_| {
>         println!("Custom panic!");
>         old_hook(); // <--- CALL IT here
>     }));
> ```

Indeed, this is safer to recap the original panic behavior. I fixed it in f619f18.